src/main/java/com/mixpanel/android/viewcrawler/ViewVisitor.java

```diff
@@ -204,73 +204,42 @@ public LayoutUpdateVisitor(List<Pathfinder.PathElement> path, LayoutRule args,
 
         @Override
         public void cleanup() {
-            for (Map.Entry<View, LayoutRule> original:mOriginalValues.entrySet()) {
+            // TODO find a way to optimize this.. remove this visitor and trigger a re-layout??
+            for (Map.Entry<View, int[]> original:mOriginalValues.entrySet()) {
                 final View changedView = original.getKey();
-                final LayoutRule originalValue = original.getValue();
-                try {
-                    setLayout(changedView, originalValue.verb, originalValue.anchor);
-                } catch (CantVisitException e) {
-                    ; // shouldn't reach here
+                final int[] originalValue = original.getValue();
+                RelativeLayout.LayoutParams originalParams = (RelativeLayout.LayoutParams) changedView.getLayoutParams();
```

Comment:
Let's declare `originalParams` final (since it's only assigned once in the block)